platform/util/platform-util/src/main/java/org/codice/ddf/platform/util/properties/PropertiesLoader.java

```diff
@@ -108,108 +126,140 @@ public Properties loadProperties(String propertiesFile) {
+      } else {
+        properties =
+            PropertiesLoaderUtils.loadAllProperties(
+                propertiesFile, PropertiesLoader.class.getClassLoader());
       }
-      properties = filtered;
+    } catch (IOException e) {
+      LOGGER.debug("Unable to load properties using default Spring properties loader.", e);
+    }
+    return properties;
+  }
 
+  /**
+   * Try loading the properties directly from the file system. If the properties file has a
+   * fully-qualified absolute path (which is what the blueprint file should specify) then it can be
+   * loaded directly, using this method. Otherwise the path will be considered relative and attempts
+   * will be made with {@code karaf.home} and {@code ddf.home} property values prepended to the
+   * original path.
+   */
+  @SuppressWarnings("squid:S1172" /* Used in bi-function */)
+  @VisibleForTesting
+  static Properties attemptLoadWithFileSystem(String propertiesFile, ClassLoader classLoader) {
+    LOGGER.debug("Attempting to load properties from file system: {}", propertiesFile);
+    Properties properties = new Properties();
+
+    String karafHome = System.getProperty("karaf.home");
+    String ddfHome = System.getProperty("ddf.home");
+
+    File propFile;
+    AbsolutePathResolver absPath = new AbsolutePathResolver(propertiesFile);
+    if (StringUtils.isNotBlank(karafHome)) {
+      propFile = new File(absPath.getPath(karafHome));
+    } else if (StringUtils.isNotBlank(ddfHome)) {
+      propFile = new File(absPath.getPath(ddfHome));
     } else {
-      LOGGER.debug("Properties file must not be null.");
+      propFile = new File(propertiesFile);
+    }
+
+    if (propFile.exists()) {
+      try (InputStreamReader reader =
+          new InputStreamReader(new FileInputStream(propFile), StandardCharsets.UTF_8)) {
+        properties.load(reader);
+      } catch (FileNotFoundException e) {
+        LOGGER.debug("Could not find properties file: {}", propFile.getAbsolutePath(), e);
+      } catch (IOException e) {
+        LOGGER.debug("Error reading properties file: {}", propFile.getAbsolutePath(), e);
+        properties.clear();
+      }
+    } else {
+      LOGGER.debug("Could not find properties file: {}", propFile.getAbsolutePath());
     }
 
     return properties;
   }
+
+  /** Try loading the properties using Java's resource loading facilities. */
+  @SuppressWarnings("squid:S1172" /* Used in bi-function */)
+  @VisibleForTesting
+  static Properties attemptLoadAsResource(String propertiesFile, ClassLoader classLoader) {
+    LOGGER.debug("Attempting to load properties as a resource: {}", propertiesFile);
+    InputStream ins = PropertiesLoader.class.getResourceAsStream(propertiesFile);
+    Properties properties = new Properties();
+    if (ins != null) {
+      try {
+        properties.load(ins);
+        ins.close();
+      } catch (IOException e) {
+        LOGGER.debug("Unable to load properties: {}", propertiesFile, e);
+        properties.clear();
```

Comment:
❗️ Once you remove the call to `close` above, you won't need to clear the properties any longer.